### PR TITLE
feat(UI): Add the option for chapter titles and dividers on the seek bar

### PR DESCRIPTION
**Pull request by Trail3lazer**
Closes https://github.com/shaka-project/shaka-player/issues/3597

### CODE REVIEW

**Comment by avelad** [comment]
@Trail3lazer I’m on vacation the next week, but I’ll review when back! Thanks!

**Comment by absidue** [comment]
Looks like you've tied the chapters to the current subtitle/caption language, how will that work if the video doesn't have any subtitles/captions or when they are disabled or if the chapters are only available in a single language but a subtitle track in a different language is active?

Probably better to either hook it up to the UI language which will always be available or create a separate setting for it.

File: docs/tutorials/ui-customization.md [hunk 1]
```diff
@@ -96,6 +96,7 @@ The following buttons can be added to the overflow menu:
 * remote: adds a button that opens a Remote Playback dialog. The button is visible only if the
   browser supports Remote Playback API.
 * Statistics: adds a button that displays statistics of the video.
+* displayChapters: tick marks between chapters and labels that appear on hover.
```
Comment (avelad) [inline docs/tutorials/ui-customization.md:99]: This configuration does not have to exist, in the same way as we do with thumbnails. Please, remove this config.

File: docs/tutorials/ui-customization.md [hunk 2]
```diff
@@ -171,7 +172,7 @@ const config = {
   'seekBarColors': {
     base: 'rgba(255, 255, 255, 0.3)',
     buffered: 'rgba(255, 255, 255, 0.54)',
-    played: 'rgb(255, 255, 255)',
+    played: 'rgb(255, 255, 255)'
```
Comment (avelad) [inline docs/tutorials/ui-customization.md:175]: Revert this change, it's not necessary

File: test/ui/ui_unit.js [hunk 3]
```diff
@@ -880,7 +880,6 @@ describe('UI', () => {
     });
   });
 
-
```
Comment (avelad) [inline test/ui/ui_unit.js:883]: Revert this change, it's not necessary

File: ui/seek_bar.js [hunk 4]
```diff
@@ -407,7 +410,6 @@ shaka.ui.SeekBar = class extends shaka.ui.RangeElement {
             'linear-gradient(' + gradient.join(',') + ')';
   }
 
-
```
Comment (avelad) [inline ui/seek_bar.js:410]: Revert this change, it's not necessary

File: ui/seek_bar.js [hunk 5]
```diff
@@ -418,7 +420,6 @@ shaka.ui.SeekBar = class extends shaka.ui.RangeElement {
     return color + ' ' + (fract * 100) + '%';
   }
 
-
```
Comment (avelad) [inline ui/seek_bar.js:421]: Revert this change, it's not necessary

File: ui/seek_bar.js [hunk 6]
```diff
@@ -437,7 +438,6 @@ shaka.ui.SeekBar = class extends shaka.ui.RangeElement {
     }
   }
 
-
```
Comment (avelad) [inline ui/seek_bar.js:440]: Revert this change, it's not necessary

File: ui/seek_bar.js [hunk 7]
```diff
@@ -461,7 +461,6 @@ shaka.ui.SeekBar = class extends shaka.ui.RangeElement {
     this.bar.ariaLabel = this.localization.resolve(shaka.ui.Locales.Ids.SEEK);
   }
 
-
```
Comment (avelad) [inline ui/seek_bar.js:464]: Revert this change, it's not necessary

File: ui/seek_bar.js [hunk 8]
```diff
@@ -542,7 +541,6 @@ shaka.ui.SeekBar = class extends shaka.ui.RangeElement {
     this.thumbnailContainer_.style.top = -(height - offsetTop) + 'px';
   }
 
-
```
Comment (avelad) [inline ui/seek_bar.js:545]: Revert this change, it's not necessary

File: ui/seek_bar.js [hunk 9]
```diff
@@ -574,7 +572,6 @@ shaka.ui.SeekBar = class extends shaka.ui.RangeElement {
     return imageTracks[0];
   }
 
-
```
Comment (avelad) [inline ui/seek_bar.js:577]: Revert this change, it's not necessary

File: ui/seek_bar.js [hunk 10]
```diff
@@ -607,7 +799,6 @@ shaka.ui.SeekBar = class extends shaka.ui.RangeElement {
   }
 };
 
-
```
Comment (avelad) [inline ui/seek_bar.js:610]: Revert this change, it's not necessary

File: ui/seek_bar.js [hunk 11]
```diff
@@ -583,6 +580,201 @@ shaka.ui.SeekBar = class extends shaka.ui.RangeElement {
...
+    */
+    const chaptersEqual = (a, b) => {
+      return !(a && b) ||
+     (a.id === b.id && a.title === b.title &&
+     a.startTime === b.startTime && a.endTime === b.endTime);
+    };
+
+    /** @type {function(): void} */
+    const handleChapterTrackChange = () => {
+      const nextLanguage = this.player.getCurrentTextLanguage() || 'und';
```
Comment (avelad) [inline ui/seek_bar.js:608]: You have to use the language of the UI not of a text track that may or may not exist.

**Review by avelad — CHANGES_REQUESTED** [review]

File: ui/seek_bar.js (hunk 11 above)
Comment (Trail3lazer) [reply ui/seek_bar.js:608]: Where do I get the language of the UI? I don't see a config value or property containing it.

**Comment by Trail3lazer** [comment]
@avelad, I removed the changes you recommended and the unnecessary configuration value. Sorry, I was unavailable to update this until now. Did you merge the changes already with a different PR?
